src/Compilers/CSharp/Portable/Lowering/SpillSequenceSpiller.cs

```diff
@@ -1132,13 +1132,16 @@ public override BoundNode VisitConditionalOperator(BoundConditionalOperator node
             }
             else
             {
-                var tmp = _F.SynthesizedLocal(node.Type, kind: SynthesizedLocalKind.Spill, syntax: _F.Syntax);
+                var isRef = node.IsRef;
+                var tmp = _F.SynthesizedLocal(node.Type, kind: SynthesizedLocalKind.Spill, syntax: _F.Syntax,
+                    refKind: isRef ? node.GetRefKindEx() : RefKind.None
```

Comment:
It looks like this change introduces a compiler crash for a scenario that worked fine before:
```
        [Fact]
        public void Test1()
        {
            var src1 = @"
using System.Threading.Tasks;

class C
{
    int F;

    static async Task Test(bool b)
    {
        Test(b ? ref (await GetC()).F : ref (await GetC()).F, await GetC());
    }

    static void Test(int x, C y) {}

    static async Task<C> GetC()
    {
        await Task.Yield();
        return new C();
    }
}
";
            var comp1 = CreateCompilation(src1, options: TestOptions.ReleaseDll);
            comp1.VerifyEmitDiagnostics();
        }
```

This scenario also crashes compiler:
```
        [Fact]
        public void Test2()
        {
            var src1 = @"
using System.Threading.Tasks;

class C
{
    int F;

    static async Task Test(bool b)
    {
        Test(ref b ? ref (await GetC()).F : ref (await GetC()).F, await GetC());
    }

    static void Test(ref int x, C y) {}

    static async Task<C> GetC()
    {
        await Task.Yield();
        return new C();
    }
}
";
            var comp1 = CreateCompilation(src1, options: TestOptions.ReleaseDll);
            comp1.VerifyEmitDiagnostics();
        }
```

Reply:
I suggest trying an alternative rewrite here, a rewrite that doesn't introduce ref locals. Instead of creating a local for result, creating a local for condition might work:
```
<condition side-effects>
bool conditionValue = condition;
if (conditionValue)
{
     <consequence side-effects>
} 
else
{
     <alternative side-effects>
}   

return conditionValue ? [ref] consequence : [ref] alternative;
```